pkg/cli/validate_test.go

```diff
@@ -0,0 +1,250 @@
+	token := testTokenBuilder(t)
+	if err := jvspb.SetJustifications(token, []*jvspb.Justification{
+		{
+			Category: "explanation",
+			Value:    "test",
+		},
+		{
+			Category: "foo",
+			Value:    "bar",
+		},
+	}); err != nil {
+		t.Fatalf("failed to set justifications in token: %v", err)
+	}
+	signedToken := testSignToken(t, token, privateKey, keyID)
+	breakglassToken, err := jvspb.CreateBreakglassToken(token, "prod is down")
+	if err != nil {
+		t.Fatalf("failed to build breakglass token: %v", err)
+	}
+
+	cases := []struct {
+		name   string
+		config *config.CLIConfig
+		args   []string
+		pipe   bool
+		expOut string
+		expErr string
+	}{
+		{
+			name:   "too_many_args",
+			args:   []string{"foo"},
+			expErr: `accepts 0 arg(s)`,
+		},
+		{
+			name:   "missing_token",
+			args:   nil,
+			expErr: `"token" not set`,
+		},
+		{
+			name: "invalid_token",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args:   []string{"-t=invalid token"},
+			expErr: `failed to parse token headers`,
+		},
+		{
+			name: "signed",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args: []string{"-t", signedToken},
+			expOut: `-----BREAKGLASS-----
+false
+
+----JUSTIFICATION----
+explanation  "test"
+foo          "bar"
+
+---STANDARD CLAIMS---
+aud  ["dev.abcxyz.jvs"]
+iat  "1970-01-01T00:00:00Z"
+iss  "jvsctl"
+jti  "test-jwt"
+nbf  "1970-01-01T00:00:00Z"
+sub  "jvsctl"
+`,
+		},
+		{
+			name: "breakglass",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args: []string{"-t", breakglassToken},
+			expOut: `-----BREAKGLASS-----
+true
+
+----JUSTIFICATION----
+breakglass  "prod is down"
+
+---STANDARD CLAIMS---
+aud  ["dev.abcxyz.jvs"]
+iat  "1970-01-01T00:00:00Z"
+iss  "jvsctl"
+jti  "test-jwt"
+nbf  "1970-01-01T00:00:00Z"
+sub  "jvsctl"
+`,
+		},
+		{
+			name: "signed_from_pipe",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args: []string{"-t", "-"},
+			pipe: true,
+			expOut: `-----BREAKGLASS-----
+false
+
+----JUSTIFICATION----
+explanation  "test"
+foo          "bar"
+
+---STANDARD CLAIMS---
+aud  ["dev.abcxyz.jvs"]
+iat  "1970-01-01T00:00:00Z"
+iss  "jvsctl"
+jti  "test-jwt"
+nbf  "1970-01-01T00:00:00Z"
+sub  "jvsctl"
+`,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			cmd := newValidateCmd(tc.config)
+			var stdout string
+			var gotErr error
+			if tc.pipe {
+				stdout, _, gotErr = testExecuteCommandStdin(t, cmd, strings.NewReader(signedToken), tc.args...)
+			} else {
+				stdout, _, gotErr = testExecuteCommand(t, cmd, tc.args...)
+			}
+			if diff := testutil.DiffErrString(gotErr, tc.expErr); diff != "" {
+				t.Fatal(diff)
+			}
+			if gotErr != nil {
+				return
+			}
+
+			if diff := cmp.Diff(tc.expOut, stdout); diff != "" {
+				t.Errorf("output: diff (-want, +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func testTokenBuilder(tb testing.TB) jwt.Token {
```

Comment:
Can you file a follow-up bug - we should probably expose this as a top-level helper in the api package.

Reply:
#142 created.